sov-modules/sov-modules-impl/integration-tests/tests/demo/basic_runtime.rs

```diff
@@ -68,186 +67,189 @@ pub struct Runtime<C: Context> {
-            assert!(result.is_ok())
-        }
-    }
-
-    // Freeze
-    {
-        let freeze_message = example_election::call::CallMessage::<C>::FreezeElection;
-
-        let serialized_message = Runtime::<C>::encode_election_call(freeze_message);
-        let module = Runtime::<C>::decode_call(&serialized_message).unwrap();
-
-        let result = module.dispatch_call(storage.clone(), &admin_context);
-        assert!(result.is_ok())
-    }
-}
-
-fn call_value_setter_module<C: Context<PublicKey = MockPublicKey>>(storage: &C::Storage) {
-    let sender = MockPublicKey::try_from("admin").unwrap();
-
-    let admin_context = C::new(sender);
-
-    // Set new value
-    let new_value = 99;
-    {
-        let set_value_msg = example_value_setter::call::CallMessage::DoSetValue(
-            example_value_setter::call::SetValue { new_value },
-        );
-
-        let serialized_message = Runtime::<C>::encode_value_setter_call(set_value_msg);
-        let module = Runtime::<C>::decode_call(&serialized_message).unwrap();
-        let result = module.dispatch_call(storage.clone(), &admin_context);
-
-        assert!(result.is_ok())
-    }
+struct Message {
+    sender: MockPublicKey,
+    data: Vec<u8>,
 }
 
-fn query_election_returns_correct_result(storage: JmtStorage) -> bool {
-    let serialized_message = QueryGenerator::generate_query_election_message();
-    let module = Runtime::<MockContext>::decode_query(&serialized_message).unwrap();
-
-    let query_response = module.dispatch_query(storage);
-
-    let response: example_election::query::Response =
-        serde_json::from_slice(&query_response.response).unwrap();
+fn execute<C: Context<PublicKey = MockPublicKey>>(storage: &C::Storage, message: Message) {
+    let module = Runtime::<C>::decode_call(&message.data).unwrap();
+    let context = Context::new(message.sender);
+    let result = module.dispatch_call(storage.clone(), &context);
 
-    response
-        == example_election::query::Response::Result(Some(Candidate {
-            name: "candidate_2".to_owned(),
-            count: 3,
-        }))
+    assert!(result.is_ok())
 }
 
-fn query_value_setter_returns_correct_result(storage: JmtStorage) -> bool {
-    let serialized_message = QueryGenerator::generate_query_value_setter_message();
-    let module = Runtime::<MockContext>::decode_query(&serialized_message).unwrap();
-
-    let new_value = 99;
+fn check_query(query: Vec<u8>, expected_response: &str, storage: JmtStorage) {
+    let module = Runtime::<MockContext>::decode_query(&query).unwrap();
     let query_response = module.dispatch_query(storage);
-    let response: example_value_setter::query::Response =
-        serde_json::from_slice(&query_response.response).unwrap();
 
-    response
-        == example_value_setter::query::Response {
-            value: Some(new_value),
-        }
+    let response = str::from_utf8(&query_response.response).unwrap();
+    assert_eq!(response, expected_response)
```

Comment:
(non-blocking): I think it would be preferable to decode the json response here instead of checking string equality. That way, we'll be robust to changes in the default json serialization behavior

Reply:
The problem is that every module defines its own response type, so if we want to keep the function generic, we have to work with strings or add generics. Since it is not blocking, I will merge it, and we can catch up tmr.